### ISSUE TITLE
Hung in IE8 with jQuery 1.8

### ISSUE THREAD

**Issue by pisi** [pisi/Reel#140]
It appears something has changed in how `$.unique()` works and it now fails (or endlessly loops to be precise) with multiple `document`s. It is filed as a [potential bug](http://bugs.jquery.com/ticket/12309) of jQuery 1.8.

These two lines causes IE8 to hang big time:

``` js
try{ pools= $.unique(pool.add(window.top.document)) }
```

Put aside the in my opinion correct expectation for jQuery to be able to decide whether two `document` objects are unique or not (afterall jQuery works just fine with `document`, right?), I'm thinking about making the comparison myself to work around this:

``` js
try{ if (pool[0] != top.document) pools= pool.add(top.document) }
```

It should have the exact same effect without the need for `$.unique()`.

**Comment by pisi** [comment]
Ok, so as was [explained](http://bugs.jquery.com/ticket/12309#comment:3) to me, what I have there was actually a misuse of `$.unique()`, so that is a big no no for our purpose. I have big hopes for the workaround outlined above. Will test and see.

**Comment by pisi** [comment]
For those, who want to give the workaround, it is available in [fix-pools branch](/pisi/Reel/tree/fix-pools). If you do, please let me know if it resolved your issue or not. Thanks!

**Comment by pisi** [comment]
Fix for this just made it into the [development branch](/pisi/Reel/tree/development)

**Comment by pisi** [comment]
I've just verified this working in:
- IE 8 and 9 on Win7
- IE 6 and 7 on WinXP

**Comment by pisi** [comment]
Merged into [master](/pisi/Reel/tree/master) and **included in today's [version 1.2.1](/pisi/Reel/tree/v1.2.1)** release.
